### PR TITLE
LibJS: Temporal spec updates, once again

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 1]
```diff
@@ -1838,4 +1838,62 @@ ThrowCompletionOr<Object*> prepare_temporal_fields(GlobalObject& global_object,
...
+    if (operation == DifferenceOperation::Since) {
+        // a. Set roundingMode to ! NegateTemporalRoundingMode(roundingMode).
+        rounding_mode = negate_temporal_rounding_mode(rounding_mode);
+    }
+
+    // 11. Let maximum be ! MaximumTemporalDurationRoundingIncrement(smallestUnit).
+    auto maximum = maximum_temporal_duration_rounding_increment(*smallest_unit);
+
+    // 12. Let roundingIncrement be ? ToTemporalRoundingIncrement(options, maximum, false).
+    auto rounding_increment = TRY(to_temporal_rounding_increment(global_object, *options, maximum.has_value() ? Optional<double> { *maximum } : Optional<double> {}, false));
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1887]:
```suggestion
    auto rounding_increment = TRY(to_temporal_rounding_increment(global_object, *options, Optional<double>(maximum), false));
```

**Review by IdanHo — COMMENTED** [review]
Nice, just a couple o' nitpicks

**Comment by Lubrsi** [comment]
Just a quick look at the commit names :^):
`LibJS: Assert AddTime is called with a valid time in AddTime` -> `LibJS: Assert AddTime is called with a valid time`

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1887]: I forgot about this, very cool!

**Comment by linusg** [comment]
Thanks both!

**Pull request merged**